src/components/SignUp/index.jsx

```diff
@@ -1,13 +1,67 @@
 import React from "react";
 import { Container, Row, Col } from "react-bootstrap";
+import { Link, useHistory } from "react-router-dom";
 import SignUpForm from "../SignUpForm";
 import womenSupportEachOther from "../../images/womenSupportEachOther.png";
 import googleIcon from "../../images/icons/googleIcon.svg";
 import twitterIcon from "../../images/icons/twitterIcon.svg";
 import facebookIcon from "../../images/icons/facebookIcon.svg";
+import firebase from "../../firebaseConfig";
+import { AuthContext } from "../../auth/Authentication";
 import "./index.scss";
 
 export default function SignUp() {
+  const history = useHistory();
+
+  const signUpSucceed = () => {
+    history.push("/");
+  };
+
+  const signUpFailed = (error) => {
+    alert(error);
+  };
+  const handleSignUp = React.useCallback(
+    async (data) => {
+      const { email, password } = data;
+
+      try {
+        await firebase.auth().createUserWithEmailAndPassword(email, password);
+        signUpSucceed();
+      } catch (error) {
+        if (error.code === "auth/email-already-in-use") {
+          return "auth/email-already-in-use";
+        }
+        signUpFailed(error);
+      }
+      return "succeed";
+    },
+    [history]
+  );
+
+  const handleSignUpWithProvider = (providerName) => {
+    let provider;
+    switch (providerName) {
+      case "google":
+        provider = new firebase.auth.GoogleAuthProvider();
+        break;
+      case "facebook":
+        provider = new firebase.auth.FacebookAuthProvider();
+        break;
+      default:
```

Comment:
shouldn't you add default?

Reply:
there are not defaut, that is to mute esLint :/

Reply:
for twitter, I created a developer account an waiting to be accepted, they are three cases